### DISCUSSION TITLE
How to align text within a selectable label?

### DISCUSSION THREAD

**Discussion by jannistpl** [emilk/egui#3868] [Q&A] [answered]
Is it possible to adjust the alignment of the text in a selectable label? If a size is specified using `ui.add_sized()`, the text will always be displayed in the center of the selectable label. However, I would like the text to start at the left edge.

Let's take the following code as an example:
```rust
egui::Window::new("Playground").show(ctx, |ui| {
    ui.add_sized([ui.available_width(), 0.0], egui::SelectableLabel::new(true, "Some label"));
});
```

This will generate the following UI:
![Screenshot_20240122_215000](https://github.com/emilk/egui/assets/55352293/9093fb60-bd69-4f73-8dc5-30569367c2e8)

The label is as wide as the window and the text is displayed in the center of the label.

I've already tried the different layout options, but I haven't made any progress yet. I didn't find anything about it in the documentation either.

This should also apply to the buttons.

Hope there is a solution for this! Thanks in advance :)

**Comment by YgorSouza** [comment, marked as answer]
You can create a [Layout](https://docs.rs/egui/0.25.0/egui/struct.Layout.html#) with the direction and alignment that you want:

```rs
ui.with_layout(Layout::top_down_justified(egui::Align::LEFT), |ui| {
    ui.selectable_label(true, "Some label");
    ui.button("Some button");
});
```

![image](https://github.com/emilk/egui/assets/43298013/5924460d-7d34-4476-a188-ca7a9eebaf60)


In fact, if you look at the source for [add_sized](https://docs.rs/egui/0.25.0/src/egui/ui.rs.html#1073-1082), all it does is create a layout with centered and justified, then add the widget inside it. Other ui methods like [vertical](https://docs.rs/egui/0.25.0/src/egui/ui.rs.html#1935-1937) and [horizontal](https://docs.rs/egui/0.25.0/src/egui/ui.rs.html#1845-1847) are also just shorthands for the most common layouts.

**Reply by jannistpl** [reply]
This works perfectly! Thank you :)
